airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -370,11 +370,11 @@ def find_pod(
             label_selector=label_selector,
         ).items
 
-        pod = None
+        pod: Optional[k8s.V1Pod] = None
         num_pods = len(pod_list)
         if num_pods > 1:
             raise AirflowException(f'More than one pod running with labels {label_selector}')
-        elif num_pods == 1:
+        elif num_pods == 1 and pod_list[0].metadata.name.startswith(self.name):
             pod = pod_list[0]
             self.log.info("Found matching pod %s with labels %s", pod.metadata.name, pod.metadata.labels)
             self.log.info("`try_number` of task_instance: %s", context['ti'].try_number)
```

Comment:
We should also log something if a pod is found but under the wrong name so the user can understand the context better when debugging.